controller.go

```diff
@@ -189,20 +192,30 @@ func (c *healthCheckController) disableStickyFailingCategories(categories map[st
 		for _, serviceName := range category.services {
 			for _, healthCheck := range healthChecks {
 				if healthCheck.Name == serviceName && !healthCheck.Ok {
-					infoLogger.Printf("Sticky category [%s] is unhealthy, disabling it.", category.name)
-					category.isEnabled = false
-					categories[catIndex] = category
-
-					err := c.healthCheckService.updateCategory(category.name, false)
-					if err != nil {
-						errorLogger.Printf("Cannot disable sticky category with name %s. Error was: %s", category.name, err.Error())
+					c.stickyCategoriesFailedServices[serviceName]++
+					infoLogger.Printf("Sticky category [%s] is unhealthy -- check %v/%v.", category.name, c.stickyCategoriesFailedServices[serviceName], category.failureThreshold)
+
+					if c.isCategoryThresholdExceeded(serviceName, category.failureThreshold) {
+						infoLogger.Printf("Sticky category [%s] is unhealthy, disabling it.", category.name)
+						category.isEnabled = false
+						categories[catIndex] = category
+
+						err := c.healthCheckService.updateCategory(category.name, false)
+						if err != nil {
+							errorLogger.Printf("Cannot disable sticky category with name %s. Error was: %s", category.name, err.Error())
+						}
+						c.stickyCategoriesFailedServices[serviceName] = 0
```

Comment:
Maybe this should be reset only after we successfully disabled the category? If there's a blip in K8S API for example, it will take another 3 attempts (or whatever the threshold is) to try disabling the category again.